.github/workflows/conda-python-test.yaml

```diff
@@ -0,0 +1,160 @@
+on:
+    workflow_call:
+      inputs:
+        build_type:
+          required: true
+          type: string
+        branch:
+          type: string
+        date:
+          type: string
+        sha:
+          type: string
+        repo:
+          type: string
+        script:
+          type: string
+          default: "ci/test_conda_python.sh"
+        run_codecov:
+          type: boolean
+          default: false
+        matrix_filter:
+          type: string
+          default: "."
+        container-options:
+          required: false
+          type: string
+          default: "-e _NOOP"
+
+defaults:
+    run:
+      shell: bash
+
+permissions:
+    actions: read
+    checks: none
+    contents: read
+    deployments: none
+    discussions: none
+    id-token: write
+    issues: none
+    packages: read
+    pages: none
+    pull-requests: read
+    repository-projects: none
+    security-events: none
+    statuses: none
+
+jobs:
+  compute-matrix:
+    runs-on: ubuntu-latest
+    env:
+      BUILD_TYPE: ${{ inputs.build_type }}
+    outputs:
+      MATRIX: ${{ steps.compute-matrix.outputs.MATRIX }}
+    steps:
+      - name: Validate Test Type
+        run: |
+          if [[ "$BUILD_TYPE" != "pull-request" ]] && [[ "$BUILD_TYPE" != "nightly" ]]; then
+              echo "Invalid build type! Must be 'nightly' or 'pull-request'."
+              exit 1
+          fi
+      - name: Compute Python Test Matrix
+        id: compute-matrix
+        run: |
+          set -eo pipefail
+
+          # Please keep the matrices sorted in ascending order by the following:
+          #
+          #     [ARCH, PY_VER, CUDA_VER, LINUX_VER, GPU, DRIVER]
+          #
+          export MATRICES="
+            pull-request:
+              # amd64
+              - { ARCH: 'amd64', PY_VER: '3.9',  CUDA_VER: '11.8.0', LINUX_VER: 'rockylinux8', GPU: 'v100', DRIVER: 'earliest' }
```

Comment:
I recommend aligning with the most current matrix. https://github.com/rapidsai/shared-workflows/blob/branch-25.04/.github/workflows/conda-python-tests.yaml#L74

Specifically, please use `l4` instead of `v100` GPUs for amd64 jobs. We have a lot of `l4` capacity but limited `v100` capacity on the new NVKS cluster. ARM nodes are all `a100` for now but will eventually include `l4` capacity.

Reply:
I would also recommend making sure to hit some "endpoints" of the supported platforms.
- Do you want to test the oldest/newest OS (`rockylinux8` has the oldest glibc in our CI images, `ubuntu24.04` has the newest glibc).
- Use oldest and newest supported Python (3.10/3.13), CUDA (11.8.0/12.8.0), driver (earliest/latest), etc.